src/CodeStyle/Tools/Program.cs

```diff
@@ -0,0 +1,302 @@
+                return 1;
+            }
+
+            var language = args[0];
+            var outputDir = args[1];
+            var targetsFileName = args[2];
+            var assemblyList = args[3].Split(new[] { ';' }, StringSplitOptions.RemoveEmptyEntries).ToList();
+
+            CreateGlobalConfigFiles(language, outputDir, assemblyList);
+            CreateTargetsFile(language, outputDir, targetsFileName);
+            return 0;
+        }
+
+        private static void CreateGlobalConfigFiles(string language, string outputDir, List<string> assemblyList)
+        {
+            Debug.Assert(language is "CSharp" or "VisualBasic");
+            var languageForGetAnalyzers = language == "CSharp" ? LanguageNames.CSharp : LanguageNames.VisualBasic;
+
+            var allRulesById = new SortedList<string, DiagnosticDescriptor>();
+            foreach (var assembly in assemblyList)
+            {
+                var analyzerFileReference = new AnalyzerFileReference(assembly, AnalyzerAssemblyLoader.Instance);
+                analyzerFileReference.AnalyzerLoadFailed += AnalyzerFileReference_AnalyzerLoadFailed;
+                var analyzers = analyzerFileReference.GetAnalyzers(languageForGetAnalyzers);
+
+                foreach (var analyzer in analyzers)
+                {
+                    foreach (var rule in analyzer.SupportedDiagnostics)
+                    {
+                        allRulesById[rule.Id] = rule;
+                    }
+                }
+            }
+
+            var configDir = Path.Combine(outputDir, "config");
+            foreach (var analysisMode in Enum.GetValues(typeof(AnalysisMode)))
+            {
+                CreateGlobalconfig(
+                    configDir,
+                    $"AnalysisLevelStyle_{analysisMode}.editorconfig",
+                    (AnalysisMode)analysisMode!,
+                    allRulesById);
+            }
+
+            return;
+
+            static void AnalyzerFileReference_AnalyzerLoadFailed(object? sender, AnalyzerLoadFailureEventArgs e)
+                => throw e.Exception ?? new InvalidOperationException(e.Message);
+
+            static void CreateGlobalconfig(
+                string folder,
+                string configFileName,
+                AnalysisMode analysisMode,
+                SortedList<string, DiagnosticDescriptor> sortedRulesById)
+            {
+                var text = GetGlobalconfigText(analysisMode, sortedRulesById);
+                var directory = Directory.CreateDirectory(folder);
+                var configFilePath = Path.Combine(directory.FullName, configFileName);
+                File.WriteAllText(configFilePath, text);
+                return;
+
+                // Local functions
+                static string GetGlobalconfigText(AnalysisMode analysisMode, SortedList<string, DiagnosticDescriptor> sortedRulesById)
```

Comment:
I don't know what's the preference on roslyn repo but local functions of local functions is a little hard on cognitive load.

Reply:
I personally like local functions of local functions and feel it avoids polluting private code used by local functions. Hence, they were added to roslyn-analyzers repo and I retained most of the code structure as-is during the port. I can change that if others disagree. I'll wait to hear from couple more folks.